internal/receiver/smartagentreceiver/event.go

```diff
@@ -0,0 +1,118 @@
+// Copyright 2021 Splunk, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package smartagentreceiver
+
+import (
+	"fmt"
+
+	"github.com/signalfx/golib/v3/event"
+	"go.opentelemetry.io/collector/consumer/pdata"
+)
+
+const (
+	SFxEventCategoryKey   = "com.splunk.signalfx.event_category"
+	SFxEventPropertiesKey = "com.splunk.signalfx.event_properties"
+)
+
+// eventToLog converts a SFx event to a pdata.Logs entry suitable for consumption by LogConsumer.
+// based on https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/5de076e9773bdb7617b544a57fa0a4b848cec92c/receiver/signalfxreceiver/signalfxv2_event_to_logdata.go#L27
+func eventToLog(event *event.Event) *pdata.Logs {
+	logs, lr := newLogs()
+
+	lr.SetName(event.EventType)
+
+	var unixNano int64
+	if !event.Timestamp.IsZero() {
+		unixNano = event.Timestamp.UnixNano()
+	}
+	lr.SetTimestamp(pdata.TimestampUnixNano(unixNano))
+
+	// size for event category and dimension attributes
+	attrsCapacity := 1 + len(event.Dimensions)
+	if len(event.Properties) > 0 {
+		attrsCapacity++
+	}
+	attrs := lr.Attributes()
+	attrs.InitEmptyWithCapacity(attrsCapacity)
+
+	if event.Category == 0 {
+		// This attribute must be present or SFx exporter will not know it's an event
+		attrs.InsertNull(SFxEventCategoryKey)
+	} else {
+		attrs.InsertInt(SFxEventCategoryKey, int64(event.Category))
+	}
+
+	for k, v := range event.Dimensions {
+		attrs.InsertString(k, v)
+	}
+
+	if len(event.Properties) > 0 {
+		propMapVal := pdata.NewAttributeValueMap()
+		propMap := propMapVal.MapVal()
+		propMap.InitEmptyWithCapacity(len(event.Properties))
+
+		for property, value := range event.Properties {
+			if value == nil {
```

Comment:
When the value is `nil` should we still account for the key? If I understand correctly it would be equivalent to this [line](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/5de076e9773bdb7617b544a57fa0a4b848cec92c/receiver/signalfxreceiver/signalfxv2_event_to_logdata.go#L75), no?

Reply:
It's never serialized with the final event, but I guess we could keep it for the error path? https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/exporter/signalfxexporter/translation/logdata_to_signalfxv2.go#L115.

Seems preferable to continue to drop it but log here instead of deferring to the exporter.

Reply:
Oh ok, yeah, I agree we should log here and continue in that case.